wgsl/index.bs

```diff
@@ -5085,15 +5085,32 @@ Note: The current function will not resume execution if the called function or
 any descendent called function executes a `discard` statement.
 See [[#discard-statement]].
 
-## Restrictions TODO ## {#function-restriction}
-TODO: *This is a stub*
-
-* Recursion is not permitted. (No cycle in the call graph.)
-* Function call parameters
- * Match type and number
- * Restrictions on pointers
- * Aliasing (?)
+## Restrictions ## {#function-restriction}
 
+* Recursion is not permitted.
+    That is, there must be no cycles in the call graph.
+* A [=vertex=] shader must return the `position` [=built-in output variable|built-in variable=].
+    See [[#builtin-variables]].
+* An entry point must never be the target of a [=function call=].
+* Functions must not return a [=pointer type=].
```

Comment:
Can't return a handle either.
I think this is better said as:   If a function has a return type, it must be a plain type.